### PR TITLE
expression,plan: fix some issues of ROUND func

### PR DESCRIPTION
**Pull request by zyguan**
I'm not very sure if I shall update the following codes also.

```
util/types/convert.go
303:            return strconv.FormatFloat(float64(v), 'f', -1, 32), nil
305:            return strconv.FormatFloat(float64(v), 'f', -1, 64), nil

util/types/mydecimal.go
951:    s := strconv.FormatFloat(f, 'g', -1, 64)

util/types/datum.go
747:            s = strconv.FormatFloat(d.GetFloat64(), 'f', -1, 32)
749:            s = strconv.FormatFloat(d.GetFloat64(), 'f', -1, 64)
1306:           return strconv.FormatFloat(float64(d.GetFloat32()), 'f', -1, 32), nil
1308:           return strconv.FormatFloat(float64(d.GetFloat64()), 'f', -1, 64), nil
```

### CODE REVIEW

File: plan/typeinferer.go [hunk 1]
```diff
@@ -262,6 +262,14 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 		if x.FnName.L == "abs" && tp.Tp == mysql.TypeDatetime {
 			tp = types.NewFieldType(mysql.TypeDouble)
 		}
+	case "round":
+		t := x.Args[0].GetType().Tp
+		switch t {
+		case mysql.TypeBit, mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLonglong:
```
Comment (tiancaiamao) [inline plan/typeinferer.go]: Should decimal be counted as Exact Value ?   @coocood

File: server/util.go [hunk 2]
```diff
@@ -325,9 +325,17 @@ func dumpTextValue(mysqlType uint8, value types.Datum) ([]byte, error) {
 	case types.KindUint64:
 		return strconv.AppendUint(nil, value.GetUint64(), 10), nil
 	case types.KindFloat32:
-		return strconv.AppendFloat(nil, value.GetFloat64(), 'f', -1, 32), nil
+		prec := -1
```
Comment (tiancaiamao) [inline server/util.go]: Any tests cover this function?

Comment (zyguan) [inline server/util.go]: @tiancaiamao I've added some tests, PTAL.

**Comment by tiancaiamao** [comment]
LGTM
PTAL @coocood

**Comment by coocood** [comment]
@zyguan 
`mysql.NewDecimal` should be handled too.
Rounding a decimal returns a decimal.

**Comment by coocood** [comment]
LGTM
Thank you!

**Pull request merged**